DelvUI/Interface/GeneralElements/HideHudConfig.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using DelvUI.Config;
+using DelvUI.Config.Attributes;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace DelvUI.Interface.GeneralElements
+{
+    [Serializable]
+    [Section("Misc")]
+    [SubSection("Hide Options", 0)]
+    public class HideHudConfig : PluginConfigObject, IOnChangeEventArgs
+    {
+        [Checkbox("Hide DelvUI outside of combat")]
+        [Order(5)]
+        public bool HideOutsideOfCombat = false;
+
+        [Checkbox("Hide Default Job Gauges", isMonitored = true)]
+        [Order(10)]
+        public bool HideDefaultJobGauges = false;
+
+        [Checkbox("Hide Default Castbar", isMonitored = true)]
+        [Order(15)]
+        public bool HideDefaultCastbar = false;
+
+        [Checkbox("Enable Combat Hotbars", isMonitored = true)]
+        [CollapseControl(20, 0)]
+        public bool EnableCombatActionBars = false;
+
+        [DynamicList("Hotbars Shown Only In Combat", "Hotbar 1", "Hotbar 2", "Hotbar 3", "Hotbar 4", "Hotbar 5", "Hotbar 6", "Hotbar 7", "Hotbar 8", "Hotbar 9", "Hotbar 10", isMonitored = true)]
+        [CollapseWith(0, 0)]
+        public List<string> CombatActionBars = new();
+
+        public new static HideHudConfig DefaultConfig() { return new HideHudConfig(); }
+
+        #region OnChangeEventInterface
+
+        // sending event outside of the config
+        public event EventHandler<OnChangeBaseArgs> onValueChanged;
+
+        // received events from the node
+        public void onValueChangedRegisterEvent(OnChangeBaseArgs e)
+        {
+            onValueChanged?.Invoke(this, e);
+        }
+
+        #endregion
```

Comment:
I think you should put this directly on PluginConfigObject so all config objects can benefit from this feature without having to do anything or repeat code.